### PR TITLE
Update chat frontend fixes #15

### LINKED ISSUE #15: Update Chat API calls front end

### PR DESCRIPTION
**Pull request by Azkle**
Make sure to use updateRoomAPI as core branch, because the endpoints have been set correctly there.

### CODE REVIEW

File: app/scripts/controllers/main.js [hunk 1]
```diff
@@ -26,8 +26,12 @@ angular.module('moodCatApp')
       );
       request.error($log.warn.bind($log, "Failed to fetch response"));
     }
+    this.fetchMessages = function(roomid){
+      $log.info("Fetched messages for chat " + roomid);
+      return $http.get('/api/rooms/'+roomid+'/messages')
+    }
   })
-  .controller('MainCtrl', function ($q, $scope, $timeout, soundCloudService, roomService, chatService) {
+  .controller('MainCtrl', function ($q, $scope, $timeout, $http, soundCloudService, roomService, chatService) {
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: Added unused import

File: app/scripts/controllers/main.js [hunk 2]
```diff
@@ -57,18 +61,23 @@ angular.module('moodCatApp')
     $scope.selectRoom = function selectRoom(room) {
       $scope.activeRoom = room;
       $scope.loadSong(room.song.soundCloudId);
-    }
 
-    /** CHAT **/
+      //Fetch chat
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: This is an example of a comment that should have been unnecessary if the code was properly refactored :wink: I suggest to extract lines 67 to 78 to a method/function `fetchChat()` and omit the comment.

File: app/scripts/controllers/main.js [hunk 3]
```diff
@@ -78,14 +87,13 @@ angular.module('moodCatApp')
...
         author: $scope.chatMessage.author,
-        roomId: 1
       };
 
       //Add the message to the local queue
       $scope.messages.push(message);
 
       //Send the mesage to the server
-      chatService.sendChatMessage(message);
+      chatService.sendChatMessage(message,$scope.activeRoom.id);
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: Space after `,` for readability.

**Comment by JWGmeligMeyling** [comment]
Few remarks, but the PR looks good :+1:

File: app/scripts/controllers/main.js (hunk 2 above)
Comment (Azkle) [inline app/scripts/controllers/main.js]: So, you would propose to change fetchMessages() to fetchChat(). Or shall I just delete that comment?

**Comment by JWGmeligMeyling** [comment]
The only left comment was discussed and won't be fixed. The PR is ready to merge!

**Comment by GijsWeterings** [comment]
LGTM, merging

**Pull request merged**